src/main/java/com/example/petbuddybackend/service/chat/ChatService.java

```diff
@@ -36,21 +54,161 @@ public Page<ChatMessageDTO> getChatMessages(Long chatId, String principalEmail,
                 .map(message -> convertTimeZone(message, timeZone));
     }
 
+    public ChatRoom getChatRoomById(Long chatId) {
+        return chatRepository.findById(chatId)
+                .orElseThrow(() -> NotFoundException.withFormattedMessage(chatId.toString(), CHAT));
```

Comment:
```suggestion
                .orElseThrow(() -> NotFoundException.withFormattedMessage(CHAT, chatId.toString()));
```
zamienilbym kolejnością i potem w metodzie tez